explorer/explorerroutes.go

```diff
@@ -202,6 +202,39 @@ func (exp *explorerUI) TxPage(w http.ResponseWriter, r *http.Request) {
 					tx.TicketInfo.PoolStatus = poolStatus.String()
 				}
 				tx.TicketInfo.SpendStatus = spendStatus.String()
+				tx.TicketInfo.TicketPoolSize = int64(exp.ChainParams.TicketPoolSize) * int64(exp.ChainParams.TicketsPerBlock)
+				tx.TicketInfo.TicketExpiry = int64(exp.ChainParams.TicketExpiry)
+				expirationInDays := (exp.ChainParams.TargetTimePerBlock.Hours() * float64(exp.ChainParams.TicketExpiry)) / 24
+				maturityInDay := (exp.ChainParams.TargetTimePerBlock.Hours() * float64(tx.TicketInfo.TicketMaturity)) / 24
+				tx.TicketInfo.TimeTillMaturity = ((float64(exp.ChainParams.TicketMaturity) - float64(tx.Confirmations)) / float64(exp.ChainParams.TicketMaturity)) * maturityInDay
+				ticketExpiryBlocksLeft := int64(exp.ChainParams.TicketExpiry) - tx.Confirmations
+				tx.TicketInfo.TicketExpiryDaysLeft = (float64(ticketExpiryBlocksLeft) / float64(exp.ChainParams.TicketExpiry)) * expirationInDays
+				if tx.TicketInfo.SpendStatus == "Voted" {
+					tx.TicketInfo.ShortConfirms = exp.blockData.TxHeight(tx.SpendingTxns[0].Hash) - tx.BlockHeight
+				} else if tx.Confirmations >= int64(exp.ChainParams.TicketExpiry) {
+					tx.TicketInfo.ShortConfirms = int64(exp.ChainParams.TicketExpiry)
+				} else {
+					tx.TicketInfo.ShortConfirms = tx.Confirmations
+				}
+				voteRounds := (tx.TicketInfo.ShortConfirms - tx.TicketMaturity)
+				tx.TicketInfo.BestLuck = tx.TicketInfo.TicketExpiry / int64(exp.ChainParams.TicketPoolSize)
+				tx.TicketInfo.AvgLuck = tx.TicketInfo.BestLuck - 1
+				tx.TicketInfo.VoteLuck = float64(tx.TicketInfo.BestLuck) - (float64(voteRounds) / float64(exp.ChainParams.TicketPoolSize))
+				if tx.TicketInfo.VoteLuck >= float64(tx.TicketInfo.BestLuck-(1/int64(exp.ChainParams.TicketPoolSize))) {
+					tx.TicketInfo.LuckStatus = "Perfection"
+				} else if tx.TicketInfo.VoteLuck > (float64(tx.TicketInfo.BestLuck) - 0.25) {
+					tx.TicketInfo.LuckStatus = "Very Lucky!"
+				} else if tx.TicketInfo.VoteLuck > (float64(tx.TicketInfo.BestLuck) - 0.75) {
+					tx.TicketInfo.LuckStatus = "Good Luck"
+				} else if tx.TicketInfo.VoteLuck > (float64(tx.TicketInfo.BestLuck) - 1.25) {
+					tx.TicketInfo.LuckStatus = "Normal"
+				} else if tx.TicketInfo.VoteLuck > (float64(tx.TicketInfo.BestLuck) * 0.50) {
+					tx.TicketInfo.LuckStatus = "Bad Luck"
```

Comment:
Having tested and taken a closer look here, I think we need to reconsider the luck categories proposed by @raedah here: https://github.com/decred/dcrdata/issues/375#issuecomment-369789909

In [this ticket](http://127.0.0.1:7777/tx/ef38d477563797320294e0f374b6a40b961c1cadaa95faf018fd1b4de3955a57), it says:

> LUCK | 3.6572                          final, Bad Luck                          (Best: 5 Avg: 4)

For an average of 4.0, I wouldn't consider 3.65722 to be "Bad Luck".  But this number is between 3.75 and 2.5.

I suggest we change the category boundaries to be linearly spaced from 0 to 5.  So we can just define each as a multiple of (5) best luck.  To the boundaries:

<pre> 0,  >0          1          2         3         4         <5, 5
no,bad      |   fair    |  ave.   |   good  |      great, perfect</pre>

Reply:
testnet and simnet go to 6

Reply:
"I wouldn't consider 3.65722 to be Bad Luck"
Why is that?  Its quite a ways beyond the 28.44 day point at which half the tickets have voted. Looks like it took about 38.19 days for this ticket.

Reply:
Yeah, scratch this.  Have a look at my comment with they plots and let me know what you think, please.